pkg/controller/admissionchecks/multikueue/batchjobadaptor.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package multikueue
+
+import (
+	"context"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+)
+
+type batchJobAdaptor struct{}
+
+var _ jobAdaptor = (*batchJobAdaptor)(nil)
+
+func (b *batchJobAdaptor) CreateRemoteObject(ctx context.Context, localClient client.Client, remoteClient client.Client, key types.NamespacedName, workloadName string) error {
+	localJob := batchv1.Job{}
+	err := localClient.Get(ctx, key, &localJob)
+	if err != nil {
+		return err
+	}
+
+	remoteJob := batchv1.Job{
+		ObjectMeta: cleanObjectMeta(&localJob.ObjectMeta),
+		Spec:       *localJob.Spec.DeepCopy(),
+	}
+
+	// cleanup
+	// drop the selector
+	remoteJob.Spec.Selector = nil
+	// drop the job labels in the template
+	delete(remoteJob.Spec.Template.Labels, "job-name")
```

Comment:
Declare a list of the labels as exported const and loop over them in the places using them